src/main/java/com/microsoft/sqlserver/jdbc/TimeoutPoller.java

```diff
@@ -28,7 +29,7 @@ static TimeoutPoller getTimeoutPoller() {
                     // initialize the timeout poller thread once
                     timeoutPoller = new TimeoutPoller();
                     // start the timeout polling thread
-                    Thread pollerThread = new Thread(timeoutPoller, "mssql-jdbc-TimeoutPoller");
+                    pollerThread = new Thread(timeoutPoller, timeoutPoller.getClass().getName());

```

Comment:
This is similar to the logging performance issue we had once where it was needlessly traversing the stack looking for some arbitrary name. That one had to do with method name so maybe this won't do the same thing, but is there a reason why we can't just hard-code a string for our name.